src/mlpack/methods/ann/activation_functions/poisson1_function.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file methods/ann/activation_functions/spline_function.hpp
+ * @author Himanshu Pathak
+ *
+ * Definition and implementation of Poisson one function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_POISSON1_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_POISSON1_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Poisson one function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = (x - 1) * e^-x \\
+ * f'(x) = e^-x + (1 - x) * e^-x \\
+ * @f}
+ */
+class Poisson1Function
```

Comment:
Is there a second Poisson function?

Reply:
Yes, there is Poisson 2 function also

Reply:
Ahh, okay, thanks for the clarification.